core/src/main/scala/org/bitcoins/core/crypto/bip32/BIP32ChainType.scala

```diff
@@ -0,0 +1,31 @@
+package org.bitcoins.core.crypto.bip32
+import scala.annotation.switch
+
+sealed abstract class BIP32ChainType
```

Comment:
Scaladoc, and 

```scala
sealed abstract class Bip32ChainType { 
  def index: Int
}

case object External extends Bip32ChainType { 
  override val index: Int = 0 
}
```

Reply:
Is this specified in BIP32 or is this BIP44? Let's make sure we get the names right with links to documentation

Reply:
you're right good catch